python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -40,42 +40,82 @@ def read_metadata(*args, **kwargs):
 
         return (new_meta, stats, parts, index)
 
+    @classmethod
+    def multi_support(cls):
+        # Assert that this class is CudfEngine
+        # and that multi-part reading is supported
+        return cls == CudfEngine
+
     @staticmethod
     def read_partition(
-        fs, piece, columns, index, categories=(), partitions=(), **kwargs
+        fs, pieces, columns, index, categories=(), partitions=(), **kwargs
     ):
         if columns is not None:
             columns = [c for c in columns]
         if isinstance(index, list):
             columns += index
 
-        if isinstance(piece, str):
-            path = piece
-            row_group = None
-            partition_keys = []
-        else:
-            (path, row_group, partition_keys) = piece
+        if not isinstance(pieces, list):
+            pieces = [pieces]
```

Comment:
Should this be ?
```suggestion
            pieces = list(pieces)
```
in case pieces is a numpy array or tuple

Reply:
Sorry - I know this is a bit ugly :/

We cant use `list(pieces)`, because pieces will either be a tuple, a string, a list of strings, or a list of tuples.  For newer versions of Dask, this should always be a list already.  For older versions, however, we want to convert a string into a list of strings, and a tuple into a list of tuples.